src/selectors/tags.js

```diff
@@ -14,7 +15,7 @@ export const getTagData = createSelector(
   (tagIDs, tagName, tagActive, tagEnabled) =>
     tagIDs.sort().map((id) => ({
       id,
-      name: tagName[id],
+      name: tagName[id] || prettifyName(id),
```

Comment:
This looks good. I am not sure if there would be a case where the id is undefined ? In that case prettifyName(id) would error out.

Reply:
it's a bit weird. but id is read from node tags. but the user said the filtering wasn't correct at their end. so this works is a good fallback, but not entirely correct.

Reply:
just to be sure can we add below check in prettifyName ? -
 
```python
export const prettifyName = (str) => {
  if (!str) {
    return ''; 
  }
  ...
```

Reply:
I'd like us to address this issue in the backend since you were able to reproduce it rather than now implementing this frontend fix, as it may hide the underlying backend error. Although I understand that breaking Kedro-Viz isn't ideal :)

Reply:
If you see in the screenshot in the `tag` array in validation object `id` is defined but good point we can add check in  `prettifyName `.

![image (10)](https://github.com/user-attachments/assets/42eb3761-ea70-43bb-968f-e9fe379d2a80)

Reply:
> I'd like us to address this issue in the backend since you were able to reproduce it rather than now implementing this frontend fix, as it may hide the underlying backend error. Although I understand that breaking Kedro-Viz isn't ideal :)

Yes, I agree. We should do both as we dont want to break Viz even if the backend errors. We can go ahead with this fix or introduce ErrorBoundary in frontend. For now I feel FE fix should be merged, but resolving the root cause and returning the tags would be handled in the backend fix

Reply:
> If you see in the screenshot in the `tag` array in validation object `id` is defined but good point we can add check in `prettifyName `.

Yup for this case it works. In general I think prettifyName should handle the edge case. Not related to this PR completely but since we are using the helper, it would be nice to add the check just to be sure.